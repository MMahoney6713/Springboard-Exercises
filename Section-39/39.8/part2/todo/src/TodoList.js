import React, {useState} from 'react';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';

const TodoList = () => {
    const [todoArray, setTodoArray] = useState([]);

    const addTodo = (text) => {
        setTodoArray(todoArray => ([
            ...todoArray,
            text
        ]))
    }

    const removeTodo = (text) => {
        const newTodoArray = todoArray.filter(todo => todo.text !== text);
        setTodoArray(newTodoArray);
    }

    return (
        <>
            <NewTodoForm addTodo={addTodo}/>
            <div className="TodoList">
                {todoArray.map(todo => (
                    <Todo text={todo.text} removeTodo={removeTodo} key={todo.text}/>
                ))}
            </div>
        </>
        
    )
}

export default TodoList;