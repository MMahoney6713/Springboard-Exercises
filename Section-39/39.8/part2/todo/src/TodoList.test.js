import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from './TodoList';

it("renders without crashing", () => {
    render(<TodoList />)
})

it("matches snapshot", () => {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
})

it("adds a todo when form is submitted", () => {
    const { getByLabelText, queryByText } = render(<TodoList />)

    expect(queryByText('Submit to github')).not.toBeInTheDocument();

    const textInput = getByLabelText('Todo Text:');
    const submitButton = queryByText('Add Todo')
    
    fireEvent.change(textInput, {target: {value: 'Submit to github'}});
    fireEvent.click(submitButton);

    expect(queryByText('Submit to github')).toBeInTheDocument();
})

it("removes todo when remove todo button is fired", () => {
    const { getByLabelText, queryByText } = render(<TodoList />)

    const textInput = getByLabelText('Todo Text:');
    const submitButton = queryByText('Add Todo')
    
    fireEvent.change(textInput, {target: {value: 'Submit to github'}});
    fireEvent.click(submitButton);

    expect(queryByText('Submit to github')).toBeInTheDocument();

    const removeTodoButton = queryByText('X');
    fireEvent.click(removeTodoButton);

    expect(queryByText('Submit to github')).not.toBeInTheDocument();
})