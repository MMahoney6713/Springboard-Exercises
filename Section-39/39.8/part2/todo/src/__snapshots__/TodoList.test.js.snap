// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="text"
    >
      Todo Text:
    </label>
    <input
      id="text"
      name="text"
      placeholder="Todo Description:"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
  <div
    class="TodoList"
  />
</DocumentFragment>
`;
